client/src/app/pages/applications/analysis-wizard/set-options.tsx

```diff
@@ -21,39 +21,53 @@ import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
 import { AnalysisWizardFormValues } from "./schema";
 import { HookFormPFGroupController } from "@app/shared/components/hook-form-pf-fields";
 import { StringListField } from "@app/shared/components/string-list-field";
-import { useFetchRulesets } from "@app/queries/rulesets";
 import {
+  getAllLabelsFromTarget,
   getParsedLabel,
-  getRulesetTargetList,
 } from "@app/common/CustomRules/rules-utils";
 import { DEFAULT_SELECT_MAX_HEIGHT } from "@app/Constants";
+import { useFetchTargets } from "@app/queries/targets";
+import defaultSources from "./sources";
 
 export const SetOptions: React.FC = () => {
   const { t } = useTranslation();
 
   const { watch, control, setValue, getValues } =
     useFormContext<AnalysisWizardFormValues>();
 
-  const {
-    formSources,
-    selectedFormSources,
-    formTargets,
-    formRulesets,
-    diva,
-    excludedRulesTags,
-    autoTaggingEnabled,
-  } = watch();
+  const { formLabels, diva, excludedRulesTags, autoTaggingEnabled } = watch();
 
   const [isSelectTargetsOpen, setSelectTargetsOpen] = React.useState(false);
   const [isSelectSourcesOpen, setSelectSourcesOpen] = React.useState(false);
-  const { rulesets } = useFetchRulesets();
+  const { targets } = useFetchTargets();
 
-  const allRulesetTargets = rulesets
-    .map((Ruleset) => getRulesetTargetList(Ruleset))
-    .flat();
+  const allLabelsFromTargets = targets
+    .map((target) => getAllLabelsFromTarget(target))
```

Comment:
If the attribute pluck is all you need, the helper function isn't really even needed and this could be written:
```js
    .map(({ labels }) => labels)
```

Reply:
+1 Will clean this up. No need to have this utility function around any more.